modules/tensor_mechanics/src/userobjects/CSVInterpolator.C

```diff
@@ -0,0 +1,157 @@
+//* This file is part of the MOOSE framework
```

Comment:
A couple of thoughts on this:

1. This also seems like a general MOOSE capability, rather than something specific to tensor_mechanics.
2. There is already a PiecewiseBilinear function that does a similar thing. Can you make use of that, maybe adding an option to do the offsetting you are doing here?

Reply:
The data from the 2D/3D simulation would most likely be obtained using a vector post processor. To use the current version of PiecewiseBilinear function, the data in the multiple csv files would have first have to be rewritten into a single file in a matrix form. Also the x and y values have to provided in the input file.

So, we would either have to add a python/shell script that first converts the data from multiple csv files into matrix format or add this alternate data file format to PiecewiseBilinear function.